src/com/cft/contactmerge/tests/CompareContactPartsTest.java

```diff
@@ -31,43 +30,648 @@ void doNamesMatch_Yes_IgnoreSpaces() {
     }
 
     @Test
-    void doNamesMatch_no_FirstNameInitialsDifferentLastName() {
+    void doNamesMatch_Maybe_FirstNameInitials_CaseInsensitive() {
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("John", "Doe", "j", "dOe"));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("J.", "Doe", "joHn", "doe"));
+    }
+
+    @Test
+    void doNamesMatch_Maybe_FirstNameInitials_CaseInsensitive_IgnoreSpaces() {
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch(" John  ", "  Doe ", "j ", " dOe"));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("J. ", " Doe", "  joHn ", " doe  "));
+    }
+
+    @Test
+    void doNamesMatch_Maybe_FirstNameInitials_CaseInsensitive_IgnoreSpaces_Punctuation() {
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch(" JosepHine-mc-vitty  ", "  Doe ", "j ", " dOe"));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("J. ", " Doe", "  josepHine Mc Vitty ", " doe  "));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch(" JosepHine-mc-vitty  ", "  sa-la-dEmoch ", "j ", " sa la Democh"));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("J. ", " sa La democh", "  josepHine Mc Vitty ", " Sa-la-democh  "));
+    }
+
+    @Test
+    void doNamesMatch_Maybe_FirstNameInitials_CaseInsensitive_IgnoreSpaces_Hyphenated() {
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch(" JosepHine-mc-vitty  ", "  Doe ", "j ", " dOe"));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("J. ", " Doe", "  josepHine Vitty ", " doe  "));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch(" JosepHine-mc-vitty  ", "  sa-la-dEmoch ", "j ", " Sa democh"));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("J. ", " Sa democh", "  josepHine Vitty ", " sa-la-dEmoch  "));
+    }
+
+    @Test
+    void doNamesMatch_No_FirstNameInitialsDifferentLastName() {
         assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("John", "Adams", "J", "Doe"));
         assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("J.", "Doe", "John", "Adams"));
     }
 
+    //------------------------------------------------------------------------------- Other Tests -------------------------------------------------------------------------------
+    @Test
+    void doNamesMatch_Maybe_Apostrophe() {
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("Adrianno", "D'onofio", "Adrianno", "D onofio"));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("Adrianno", "Donofio", "Adrianno", "D'onofio"));
+    }
+
+    @Test
+    void doNamesMatch_Maybe_Apostrophe_CaseInsensitive() {
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("adriaNno", "d'onOfio", "adrianNo", "d onofiO"));
```

Comment:
I'm surprised these aren't "yes" results. We want them to be, right? And one of the last name parts is greater than 3 characters so even with our compromise, it should be a "yes" and not "maybe".

Reply:
nice catch, yes... the return results for these "maybe" tests should be yes.